### PR TITLE
cluster delete rbac fix

### PR DESCRIPTION
**Pull request by Randy424**
Signed-off-by: randybrunopiverger <rbrunopi@redhat.com>

https://github.com/open-cluster-management/backlog/issues/12709

### CODE REVIEW

**Comment by Randy424** [comment]
@jamestalton 
After those last changes we hit this after destroy button is pressed:

Screen Shot 2021-07-08 at 1 39 21 PM

...though, Destroy still succeeds

**Comment by Randy424** [comment]
/unhold
working now

File: frontend/src/lib/delete-cluster.ts [hunk 1]
```diff
@@ -38,11 +40,11 @@ export function deleteCluster(cluster: Cluster, ignoreClusterDeploymentNotFound
     return {
         promise: new Promise((resolve, reject) => {
             deleteResourcesResult.promise.then((promisesSettledResult) => {
-                if (promisesSettledResult[0].status === 'rejected') {
+                if (promisesSettledResult[0]?.status === 'rejected') {
                     reject(promisesSettledResult[0].reason)
                     return
                 }
-                if (promisesSettledResult[1].status === 'rejected') {
+                if (promisesSettledResult[1]?.status === 'rejected') {
```
Comment (jamestalton) [inline frontend/src/lib/delete-cluster.ts:47]: we need to switch around the logic as it expects promisesSettledResult[1] to be for the cluster deployment and
promisesSettledResult[0] will now be for the cluster deployment

**Review by jamestalton — CHANGES_REQUESTED** [review]

**Review by jamestalton — APPROVED** [review]
lgtm

**Pull request merged**